### DISCUSSION TITLE
LSP doesn't seems to care much about extends

### DISCUSSION THREAD

**Discussion by freerig** [typescript-language-server/typescript-language-server#937] [Q&A] [answered]
Hi,
I have some types in `.svelte-kit/ambient.d.ts` that are imported in the `include` of `.svelte-kit/tsconfig.json`. `.svelte-kit/tsconfig.json` is referenced in the `extends` of the main `tsconfig.json` (in root of the project). In this configuration, the LSP says that `$app/navigation`'s type definitions aren't found (and every other `$app/...` btw).
The thing is that when I put `.svelte-kit/ambient.d.ts` in the `include` of the main `tsconfig.json` direcly, it works. (but this is not a real solution, right ?)
typescript-language-server v4.4.3
Does someone has the same issue ?

PS : already looked #175, but no...

**Comment by rchl** [comment, marked as answer]
You need to provide reproduction to get a proper answer but I'm guessing that the root `tsconfig.json` is overwriting include/files of the `tsconfig.json` in `.svelte-kit`.

The server doesn't have any handling of tsconfig of its own. It's the typescript itself that does it so consult with typescrript documentation on how this works - https://www.typescriptlang.org/tsconfig/#extends

**Reply by freerig** [reply]
Thanks for your answer

> I'm guessing that the root tsconfig.json is overwriting include/files of the tsconfig.json in .svelte-kit.

Yes, this is the problem, but I don't realy know how to solve this, since there are files that I wanna add in the main `tsconfig.json` that aren't in `.svelte-kit/tsconfig.json`. It works when I use `files` instaid of `include` in the root `tsconfig.json`, but as I understood from [the doc](https://www.typescriptlang.org/tsconfig/#files), `include` is the only way of adding a bunch of files (`src/**/*`)...

As I don't realy need to add `src/**/*` explicitly in this case, that's ok, but for other projects, I would be curious to know how to merge the includes, instead of overwriting them, when using extends. But anyway I believe that's not related to LSP...

I still believe the best configuration language is Nix.
